File: app/src/main/java/dev/chu/architecture/java/p87/MainFragmentComponent.java
package dev.chu.architecture.java.p87;

import dagger.BindsInstance;
import dagger.Subcomponent;
import dev.chu.architecture.etc.FragmentScope;

@FragmentScope
@Subcomponent(modules = MainFragmentModule.class)
public interface MainFragmentComponent {

    void inject(MainFragment mainFragment);

    @Subcomponent.Builder
    interface Builder {
        MainFragmentComponent.Builder setModule(MainFragmentModule module);
        @BindsInstance
        MainFragmentComponent.Builder setFragment(MainFragment fragment);
        MainFragmentComponent build();
    }

}

File: app/src/main/java/dev/chu/architecture/kotlin/p87/MainFragmentComponent.kt
package dev.chu.architecture.kotlin.p87

import dagger.BindsInstance
import dagger.Subcomponent
import dev.chu.architecture.etc.FragmentScope

@Subcomponent(modules = [MainFragmentModule::class])
@FragmentScope
interface MainFragmentComponent {
    fun inject(fragment: MainFragment)

    @Subcomponent.Factory
    interface Factory {
        fun create(
            module: MainFragmentModule,
            @BindsInstance fragment: MainFragment
        ): MainFragmentComponent
    }
}
